python/ray/test/cluster_utils.py

```
@@ -0,0 +1,181 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+import time
+
+import ray
+import ray.services as services
+
+logger = logging.getLogger(__name__)
+
+
+class Cluster():
+    def __init__(self, initialize_head=False, connect=False, **head_node_args):
+        """Initializes the cluster.
+
+        Arguments:
+            initialize_head (bool): Automatically start a Ray cluster
+                by initializing the head node. Defaults to False.
+            connect (bool): If `initialize_head=True` and `connect=True`,
+                ray.init will be called with the redis address of this cluster
+                passed in.
+            head_node_args (kwargs): Arguments to be passed into
+                `start_ray_head` via `self.add_node`.
+        """
+        self.head_node = None
+        self.worker_nodes = {}
+        self.redis_address = ""
+        if initialize_head:
+            self.add_node(**head_node_args)
+            if connect:
+                ray.init(redis_address=self.redis_address)
+        elif connect:
```

Maybe just raise an exception earlier on if `not initialize_head and connect`